File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        if(s.length()==0){return 0;}
        int r=0;
    List<Character> map=new ArrayList<>();
    for(int i=0;i<s.length();i++){
        // left=0;
        while(map.contains(s.charAt(i))){
            map.remove(0);
        }
        map.add(s.charAt(i));
        r=Math.max(r,map.size());
    }
    return r;
    }
}

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        d=set()
        i=0
        j=0
        maxi=0
        while j<len(s):
            while i<=j and s[j] in d:
                d.remove(s[i])
                i+=1
            else:
                d.add(s[j])
                maxi=max(maxi,j-i+1)
                j+=1
        return maxi
            